data/account/participationRegistry.go

```diff
@@ -845,10 +845,17 @@ func (db *participationDB) GetForRound(id ParticipationID, round basics.Round) (
 		return ParticipationRecordForRound{}, ErrRequestedRoundOutOfRange
 	}
 
+	// Delete all secrets except for the requested round
+	snap := result.ParticipationRecord.Voting.Snapshot()
+	otsID := basics.OneTimeIDForRound(round, result.KeyDilution)
+	snap.DeleteBeforeFineGrained(otsID, result.KeyDilution)
+	snap.DeleteAllButFineGrained(otsID)
+	result.ParticipationRecord.Voting = &snap
```

Comment:
The heart of the change is this block of code in the participation registry. Stripping out the extra stuff and seeing that the network operates properly shows that a subset of secrets may be returned without any trouble and the existing public interface doesn't require any changes.